### PR TITLE
Add list of glanceAPI support

### PR DESCRIPTION
**Pull request by fmount**
This change aligns the `openstack-operator` with the work done in the [glance-operator](https://github.com/openstack-k8s-operators/glance-operator/pull/384) to support the deployment of a list of `glanceAPI`. The logic has been improved to make sure the new `CRD` is supported.

Demo: https://asciinema.org/a/626515

- [x]  https://github.com/openstack-k8s-operators/glance-operator/pull/384

Depends-On: https://github.com/openstack-k8s-operators/data-plane-adoption/pull/221

### CODE REVIEW

File: apis/core/v1beta1/openstackcontrolplane_types.go
Comment (abays) [inline apis/core/v1beta1/openstackcontrolplane_types.go:305]: I think the samples below need to be updated to reflect this change.  i.e. they still have:

```
  glance:
    apiOverride:
      route: {}
```

Comment (fmount) [reply apis/core/v1beta1/openstackcontrolplane_types.go:305]: Yeah, updating the related sample files in my next PS.

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
@abays I think that once this patch is rebased on its dependency (https://github.com/openstack-k8s-operators/glance-operator/pull/384), we can try to update the `Depends-on` for zuul to https://github.com/openstack-k8s-operators/data-plane-adoption/pull/221 and see if the adoption job works as expected. Even though the other jobs are failing, the Glance deployment looks good, so I'll recheck this patch once the Glance one merges.

**Comment by fmount** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

File: config/samples/core_v1beta1_openstackcontrolplane_galera_network_isolation.yaml
Comment (abays) [inline config/samples/core_v1beta1_openstackcontrolplane_galera_network_isolation.yaml:1]: It looks like this file needs an update to `glance.apiOverride`

File: config/samples/core_v1beta1_openstackcontrolplane_network_isolation.yaml
Comment (abays) [inline config/samples/core_v1beta1_openstackcontrolplane_network_isolation.yaml:1]: It looks like this file also needs an update to `glance.apiOverride`

File: config/samples/core_v1beta1_openstackcontrolplane_network_isolation_ceph.yaml
Comment (abays) [inline config/samples/core_v1beta1_openstackcontrolplane_network_isolation_ceph.yaml:1]: It looks like this file also needs an update to `glance.apiOverride`

File: config/samples/core_v1beta1_openstackcontrolplane_network_isolation.yaml
Comment (fmount) [reply config/samples/core_v1beta1_openstackcontrolplane_network_isolation.yaml:1]: +1 good catch! doing that in my next iteration.

File: config/samples/core_v1beta1_openstackcontrolplane_galera_network_isolation.yaml
Comment (fmount) [reply config/samples/core_v1beta1_openstackcontrolplane_galera_network_isolation.yaml:1]: +1 good catch! doing that in my next iteration.

File: config/samples/core_v1beta1_openstackcontrolplane_network_isolation_ceph.yaml
Comment (fmount) [reply config/samples/core_v1beta1_openstackcontrolplane_network_isolation_ceph.yaml:1]: +1 good catch! doing that in my next iteration.

File: config/samples/core_v1beta1_openstackcontrolplane_galera_network_isolation.yaml
Comment (fmount) [reply config/samples/core_v1beta1_openstackcontrolplane_galera_network_isolation.yaml:1]: Done

File: config/samples/core_v1beta1_openstackcontrolplane_network_isolation_ceph.yaml
Comment (fmount) [reply config/samples/core_v1beta1_openstackcontrolplane_network_isolation_ceph.yaml:1]: Done

File: config/samples/core_v1beta1_openstackcontrolplane_network_isolation.yaml
Comment (fmount) [reply config/samples/core_v1beta1_openstackcontrolplane_network_isolation.yaml:1]: Done

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
After rechecking zuul we can see that:

1. Data Plane Adoption failure is expected, because it requires [1];
2. Tempest multinode fails, but looks like we've lost OCP in that job, even the `must-gather` is not able to retrieve any log
3. Both baremetal and crc-net-isolation jobs work as expected, and for example we can see the right CR being created [2]

@abays not sure you have extra comments on this one, but once I have the green light I'll start merging the glance dependency so we can focus to get this one in. 

[1] https://github.com/openstack-k8s-operators/data-plane-adoption/pull/221
[2] https://logserver.rdoproject.org/87/587/1a38ed3932845b92e816339a56bfe15a4cba856d/github-check/cifmw-crc-podified-edpm-baremetal/22e8e91/ci-framework-data/logs/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-c5501ac502209eda400116b0511a5cf07f0537f650b706f63006628472a846af/namespaces/openstack/crs/openstackcontrolplanes.core.openstack.org/openstack-galera-network-isolation.yaml

**Comment by fmount** [comment]
recheck

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**
